assets/js/theme/common/carousel/utils/handleImageLoad.js

```diff
@@ -0,0 +1,24 @@
+import getActiveSlideInfo from './getActiveSlideInfo';
+
+const IMAGE_ERROR_CLASS = 'is-image-error';
+const IS_ANALYZED_DATA_ATTR = 'image-load-analyzed';
+
+export default (e, carousel) => {
+    const {
+        isAnalyzedSlide,
+        $activeSlideContent,
+        $activeSlideImg,
+    } = getActiveSlideInfo(carousel, IS_ANALYZED_DATA_ATTR);
+
+    if (isAnalyzedSlide) return;
+
+    $activeSlideContent.data(IS_ANALYZED_DATA_ATTR, true);
+
+    const imageNode = $activeSlideImg[0];
+
+    if (imageNode.complete && imageNode.naturalHeight === 0) {
```

Comment:
Are there cases when imageNode could be null or undef?

Reply:
No such cases, because we take image nodes from static part of layout